src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -176,15 +176,74 @@ class ArnoldOutput : public IECore::RefCounted
 				AiNodeSetStr( m_driver.get(), AiParamGetName( fileNameParameter ), output->getName().c_str() );
 			}
 
+			IECore::StringVectorDataPtr customAttributesData = new IECore::StringVectorData();
+			std::vector<std::string> &customAttributes = customAttributesData->writable();
 			for( IECore::CompoundDataMap::const_iterator it = output->parameters().begin(), eIt = output->parameters().end(); it != eIt; ++it )
 			{
 				if( boost::starts_with( it->first.string(), "filter" ) )
 				{
 					continue;
 				}
+
+				std::string formattedData;
+				if( boost::starts_with( it->first.string(), "header:" ) )
+				{
+					if( const IECore::BoolData *data = IECore::runTimeCast<const IECore::BoolData>( it->second.get() ) )
```

Comment:
This function is becoming huuuge. How about factoring out the big list of ifs into a `formatHeaderParameter( const Data *data )` function?